pkg/provider/provider.go

```diff
@@ -98,3 +109,42 @@ func GetTestEnvironment() TestEnvironment {
 func IsOCPCluster() bool {
 	return !env.variables.NonOcpCluster
 }
+
+func WaitDebugPodReady() {
+	oc := ocpclient.NewOcpClient()
+	listOptions := metav1.ListOptions{}
+	nodes, err := oc.Coreclient.Nodes().List(context.TODO(), listOptions)
+
+	if err != nil {
+		logrus.Fatalf("Error getting node list, err:%s", err)
+	}
+
+	nodesCount := int32(len(nodes.Items))
+
+	getOptions := metav1.GetOptions{}
+	isReady := false
+	start := time.Now()
+	for !isReady && time.Since(start) < timeout {
+		daemonSet, err := oc.AppsClient.DaemonSets(daemonSetNamespace).Get(context.TODO(), daemonSetName, getOptions)
+		if err != nil && daemonSet != nil {
+			logrus.Fatal("Error getting Daemonset, please create debug daemonset")
+		}
+		if daemonSet.Status.DesiredNumberScheduled != nodesCount {
+			logrus.Fatalf("Daemonset DesiredNumberScheduled not equal to number of nodes:%d, please instantiate debug pods on all nodes", nodesCount)
+		}
+		if daemonSet.Status.DesiredNumberScheduled == daemonSet.Status.CurrentNumberScheduled && //nolint:gocritic
+			daemonSet.Status.DesiredNumberScheduled == daemonSet.Status.NumberAvailable &&
+			daemonSet.Status.DesiredNumberScheduled == daemonSet.Status.NumberReady &&
+			daemonSet.Status.NumberMisscheduled == 0 {
+			isReady = true
+		}
+		logrus.Debugf("Waiting for debug pods to be ready: %v", &daemonSet.Status)
+		time.Sleep(time.Second)
+	}
```

Comment:
Could this section be broken out into its own function and unit tested?

Reply:
fixed